src/bootstrap/src/core/build_steps/setup.rs

```diff
@@ -607,28 +674,32 @@ fn create_vscode_settings_maybe(config: &Config) -> io::Result<bool> {
         }
     };
     if should_create {
-        let path = config.src.join(".vscode");
+        let path = settings_path.as_path().parent().unwrap();
         if !path.exists() {
-            fs::create_dir(&path)?;
+            fs::create_dir(path)?;
         }
         let verb = match mismatched_settings {
             // exists but outdated, we can replace this
             Some(true) => "Updated",
             // exists but user modified, back it up
             Some(false) => {
                 // exists and is not current version or outdated, so back it up
-                let mut backup = vscode_settings.clone();
-                backup.set_extension("json.bak");
-                eprintln!("WARNING: copying `settings.json` to `settings.json.bak`");
-                fs::copy(&vscode_settings, &backup)?;
+                let mut backup = settings_path.clone();
+                backup.set_extension(editor.backup_extension());
```

Comment:
Suggestion: this can be `PathBuf::with_extension`.

Reply:
ah neat, didn't realise that was there, thanks!

Reply:
Would be even nicer to use [with_added_extension](https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.with_added_extension) but it's not stable yet so idk if it's worth the bother.